doc/adrs/011_refactor_incremental_layout.adoc

```diff
@@ -0,0 +1,231 @@
+The next step consists on considering all nodes, starting from the deepest, and to set their positions in their containers according to their siblings.
+Once all the positions & sizes of the child of a container has been set, we can set its size accordingly.
+We can then proceed and compute the position & sizes of the parent elements.
+
+==== Node labels sizes and positions
+
+===== Dependencies
+
+The node size, necessary to center the label position.
+The new text size VS the old one. This means that to only recompute labels that actually changed, we would have to keep a track of the old texts.
+As the label computation is quite trivial, for a first step it would be easier to always recompute all of them.
+
+===== Computation
+
+The Node labels size/positions can be set regardless of the other data as they are relative to the designated element.
+
+==== Edge bendpoints
+
+===== Dependencies
+
+The source/target positions & sizes.
+
+===== Computation
+
+The edge bendpoints consists on two mandatory bendpoints, the source & target anchors, and optional intermediary bendpoints. There are all stored as a whole in a list of positions for a given edge.
+As this is only a refactoring, that does not intend to solve all the issues at all, a first version can simply match what we had following ADR-008:
+
+- if an edge has moved, all intermediary bendpoints are removed
+- no intermediary bendpoints are recomputed
+- source & target anchor points to the middle of a given element
+
+==== Edge label sizes and positions
+
+===== Dependencies
+
+The new text size VS the old one.
+The edge bendpoints
+
+===== Computation
+
+As for the bendpoints, we will match the previous layout provided by ADR-008.
+
+==== Overlaps resolution
+
+At some point we need to reconsider the previous computations in order to fix overlapping elements, when possible.
+This computation can be isolated and triggered at the right moment, when all the children of a graphical container (diagram or node with children) have been rightfully computed.
+
+The point of this resolution is to move all elements that can be, in order to avoid an overlap. An element cannot be moved if it has been moved by the user, so the other elements needs to move around it.
+
+The expected feedback when an element A has been moved over an element B, is to move B along the axis of the centers of A & B, with a minimal distance between them.
+
+==== Containment resolution
+
+As for the overlap resolution, we want to avoid elements to be outside of their container.
+Any step that can move the children of a container must be done before that containment resolution.
+
+There can be two cases of invalid elements positions:
+
+- an element is on the right or at the bottom of its container: we must resize the container accordingly to make it visually contain the element
```

Comment:
container accordingly

Reply:
Fixed in https://github.com/wpiers/sirius-components/commit/62f8cf80e053d114a0f7655bd26db3318e521999